modules/library/referencing/src/test/java/org/geotools/referencing/operation/projection/GeostationarySatelliteTest.java

```diff
@@ -0,0 +1,354 @@
+        assertThat(p, is(notNullValue()));
+        geogToSphericalGeos.transform(p, p);
+        assertThat(p, is(notNullValue()));
+
+        p.setLocation(inscribed.getMinX(), inscribed.getMinY());
+        sphericalGeosToGeog.transform(p, p);
+        assertThat(p, is(notNullValue()));
+        geogToSphericalGeos.transform(p, p);
+        assertThat(p, is(notNullValue()));
+
+        // Inscribed rectangle is smaller than largest inscribing rectangle, hence ESTIMATE
+//        final double tickle = 1;
+//        expectProjectionException(new Testable() { public void test() throws Exception {
+//            p.setLocation(inscribed.getMaxX() + tickle, inscribed.getMaxY() + tickle);
+//            sphericalGeosToGeog.transform(p, p);
+//        }});
+//        expectProjectionException(new Testable() { public void test() throws Exception {
+//            p.setLocation(inscribed.getMaxX() + tickle, inscribed.getMinY() - tickle);
+//            sphericalGeosToGeog.transform(p, p);
+//        }});
+//        expectProjectionException(new Testable() { public void test() throws Exception {
+//            p.setLocation(inscribed.getMinX() - tickle, inscribed.getMaxY() + tickle);
+//            sphericalGeosToGeog.transform(p, p);
+//        }});
+//        expectProjectionException(new Testable() { public void test() throws Exception {
+//            p.setLocation(inscribed.getMinX() - tickle, inscribed.getMinY() - tickle);
+//            sphericalGeosToGeog.transform(p, p);
+//        }});
+    }
+
+    @Test
+    public void testInscribeFullDiskEstimate_Ellipsoidal() throws TransformException, FactoryException {
+
+        final Envelope2D inscribed = GeostationarySatellite.inscribeFullDiskEstimate(ellipsoidalGeosCRS);
+        assertThat(inscribed, is(notNullValue()));
+
+        final DirectPosition2D p = new DirectPosition2D();
+
+        p.setLocation(inscribed.getMaxX(), inscribed.getMaxY());
+        ellipsoidalGeosToGeog.transform(p, p);
+        assertThat(p, is(notNullValue()));
+        geogToEllipsoidalGeos.transform(p, p);
+        assertThat(p, is(notNullValue()));
+
+        p.setLocation(inscribed.getMaxX(), inscribed.getMinY());
+        ellipsoidalGeosToGeog.transform(p, p);
+        assertThat(p, is(notNullValue()));
+        geogToEllipsoidalGeos.transform(p, p);
+        assertThat(p, is(notNullValue()));
+
+        p.setLocation(inscribed.getMinX(), inscribed.getMaxY());
+        ellipsoidalGeosToGeog.transform(p, p);
+        assertThat(p, is(notNullValue()));
+        geogToEllipsoidalGeos.transform(p, p);
+        assertThat(p, is(notNullValue()));
+
+        p.setLocation(inscribed.getMinX(), inscribed.getMinY());
+        ellipsoidalGeosToGeog.transform(p, p);
+        assertThat(p, is(notNullValue()));
+        geogToEllipsoidalGeos.transform(p, p);
+        assertThat(p, is(notNullValue()));
+
+        // Inscribed rectangle is smaller than largest inscribing rectangle, hence ESTIMATE
```

Comment:
I'd intended the method to return the maximum-area inscribed rectangle from the visible ellipse in a full disk image. The tests were to verify that condition. The reality is that the projection math is a little more complicated and what is returned is the almost-maximum area inscribed rectangle.  It's also possible that an ellipse or the ellipse axes i am using are not the appropriate. I would like to address this eventually...

This method is intended to aid in coverage clipping, resampling and tiling where it's problematic if coverage bounding box can't be transformed.  This work is on my road map.